### ISSUE TITLE
Use WebGL renderer for OpenLayers VectorLayer & VectorTileLayer

### ISSUE THREAD

**Issue by spwoodcock** [hotosm/fmtm#842]
## Issue

Poor performance of geojson and vector tile layers in OpenLayers.

## Solution

Relatively new WebGL renderers for OpenLayers.
- VectorLayer renderer: https://openlayers.org/en/latest/examples/webgl-vector-layer.html
- VectorTileLayer renderer: https://openlayers.org/en/latest/examples/webgl-vector-tiles.html
(ol/layer/WebGLTile is already available for raster sources)

They should be available in ol v8.
Although the API is not fully stable yet, we can accept this.
Test the renderers to see if they solve performance issues.
(if the VectorLayer renderer works nicely, we won't have to bother testing with flatgeobuf / geoparquet for now)

**Comment by varun2948** [comment]
@spwoodcock should we go into this to implement vectorlayer with webgl as of now in its own documentation  page it says something like this so should we wait more or should we go ahead ?
![image](https://github.com/hotosm/fmtm/assets/37866666/0665ed3b-b21f-4920-ae8b-17c9c5b18fa8)

**Comment by spwoodcock** [comment]
Yeah I think it's fine to use - it's been available a little while & I doubt the API will change too much. 

Definitely worth a test at the minimum to assess performance 😁

**Comment by varun2948** [comment]
okay will work on it then

**Comment by spwoodcock** [comment]
Any functionality you end up with could be ported to the shared map components once started 🙌

**Comment by NSUWAL123** [comment]
@spwoodcock should we go to https://openlayers.org/en/latest/examples/geojson-vt.html vt layer instead of webgl as this vt layer looks more stable than the webgl one ? what's your take ?

**Comment by spwoodcock** [comment]
Not a bad idea, but GeoJSONVT is a frontend optimisation for when you are not in control of the source format.

We have full flexibility to output whatever we need, so Vector Tiles are a more efficient format (I think the author of GeoJSONVT format even says this).
It might be good to benchmark the difference between the two / do some testing if interested.

We also need to consider that we need to load both mbtiles and pmtiles, requiring Vector Tile rendering.
For the data extracts we will also use flatgeobuf (binary encoded vector data).

So in summary, I think we need to find a way to efficiently render vector tiles. This could be the WebGL renderer for OpenLayers (I think it's becoming more stable as time passes), but I am open to other suggestions.

**Comment by NSUWAL123** [comment]
so i tried with @varun2948 and we have issues regarding click feature functionality with geojson provided and the example has URL example so we are having some weird issues but trying still and will let u know if it gets done or otherwise.

**Comment by NSUWAL123** [comment]
@spwoodcock, there are issues of onclick feature with webGL so we are now gonna try layer vt

**Comment by spwoodcock** [comment]
Ok, thanks for checking 😄

How would we handle pmtile or mbtile with VT though?

**Comment by spwoodcock** [comment]
On second thought doesn't it make more sense to also use flatgeobuf for task areas?

So:

- Vector data: flatgeobuf
- Basemap data: pm/mbtiles

**Comment by spwoodcock** [comment]
We used vector tiles initially in FMTM, but have since replaced with with other formats.

We use the WebGL raster tile (for basemaps) implementation and it works fine.

Closing this as I can't see a need for improved vector tile performance for now.

The map in https://github.com/hotosm/fmtm/milestone/36 will probably be MapLibre & we may look at migrating FMTM to use MapLibre in future to better align with Drone TM.
